hoot-core/src/main/cpp/hoot/core/ops/UnconnectedWaySnapper.cpp

```diff
@@ -469,16 +470,13 @@ int UnconnectedWaySnapper::_getNodeToSnapWayInsertIndex(
   }
   LOG_VART(indexOfSecondClosestWayNodeId);
 
-  // Our way node snap index is either right after or before the closest way node.
-  int nodeToSnapInsertIndex = -1;
+  //  Our way node snap index either takes the place of the nearest, pushing it to the right, or
+  //  it is right after the closest way node.
+  int nodeToSnapInsertIndex = indexOfClosestWayNodeId;
   assert(indexOfSecondClosestWayNodeId != indexOfClosestWayNodeId);
   if (indexOfSecondClosestWayNodeId > indexOfClosestWayNodeId)
   {
-    nodeToSnapInsertIndex = indexOfClosestWayNodeId + 1;
-  }
-  else
-  {
-    nodeToSnapInsertIndex = indexOfClosestWayNodeId - 1;
+    nodeToSnapInsertIndex++;
```

Comment:
This was the big change to the `UnconnectedWaySnapper`.  The insertion index was off by one half of the time.  If the nearest node was found at index 18 and it should come before that node in the node/way ordering then it should be inserted at index 18 and shift everything to the right, not at 17 shifting 17 and 18+ to the right.  If it should come after 18 then it is incremented and stuck in before 19.